assignments/arrays.js

```diff
@@ -63,32 +63,63 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+console.log(`Car 33 is a ${inventory[32].car_year} ${inventory[32].car_make} ${inventory[32].car_model}`);
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+
+console.log(`Car 33 is a ${inventory[inventory.length - 1].car_make} ${inventory[inventory.length - 1].car_model}`);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
-console.log();
+for (let i = 0; i < inventory.length; i++) {
+    carModels.push(inventory[i].car_model);
+}
+//sorts car_model values of inventory alphabetically and console.logs the result
+console.log(carModels.sort());
+
+//verifies that alphabetically sorted array is same length as inventory array  to ensure there are no missing entries
+console.log(carModels.length === inventory.length);
+
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+
+//adds year of each inventory item to carYears
+for (let i = 0; i < inventory.length; i++) {
+    carYears.push(inventory[i].car_year);
+}
+
+//logs the array of years
+console.log(carYears);
+
+//verifies that  array is same length as inventory array  to ensure there are no missing entries
+console.log(carYears.length === inventory.length);
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars = [];
-console.log(); 
+
+for (let i = 0; i < carYears.length; i++) {
+    if (carYears[i] > 2000) {
+        oldCars.push(carYears[i]);
+    }
+}
+console.log(oldCars.length);
 
 // ==== Challenge 6 ====
-// A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
+// A buyer is interested in seeing only BMW and Audi cars within the inventory.
+// Return an array that only contains BMW and Audi cars.
+// Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi = [];
-console.log();
+for (let i = 0; i < inventory.length; i++) {
+    if (inventory[i].car_make === "BMw" || inventory[i].car_make === "Audi") {
```

Comment:
I'm only seeing audis in the console, too bad there's no spellcheck in webStorm

Reply:
I kid I kid

Reply:
Dang it. Had a lowercase m. Fixed in latest commit.

Reply:
🔥